rfd/0080-ssh-agent.md

```diff
@@ -0,0 +1,109 @@
+
+```bash
+# the next call to tsh will require you to log in again
+> tsh ls --add-keys-to-agent=only
+Enter password for Teleport user dev:
+...
+# the next call to tsh will require you to log in again
+> tsh ls
+Enter password for Teleport user dev:
+...
+```
+
+`tsh` will be updated to look for agent keys with the label `teleport:user` (or `teleport:cluster_name:user` after https://github.com/gravitational/teleport/pull/14677). If `tsh` finds an agent key, then it will be used for any certificate signing or tls/ssh handshakes during the `tsh` command execution. If no agent key is found, then `tsh` will default to using the private key stored at `~/tsh/keys/proxy.example.com/user`. 
+
+#### Using pregenerated agent keys
+
+With the strategy above, it is possible for a user to create a private key and add it to their `ssh-agent` themselves - `ssh-keygen -t rsa -C "teleport:user" -f ~/.ssh/dev`. As a result, the private key which exists outside of the `ssh-agent` would be used for `tsh` operations. 
+
+To combat this, `tsh` will not trust outside keys by default. Instead, `tsh login` will continue to create the private key itself and add it to the `ssh-agent`. Then, `tsh` will generate the user's ssh and tls certificates using the private key in memory. These certificates will be used to verify that the agent key hasn't been replaced in future calls to `tsh`.
+
+To bypass this, a user can use the new flag, `--use-agent-key="agent-key-name"`, which will enable integrations with tools like `yubikey-agent` or `pivy-agent`. This may be prefered for trusted users in a Teleport cluster, since the private key will never exist outside of the yubikey's PIV, not even in-memory during `tsh login`. 
+
+### yubikey PIV support
+
+[go-piv](https://github.com/go-piv/piv-go) can be used to integrate directly with a yubikey to create/store private keys, sign/store certificates, and perform tls and ssh handshakes.
+
+#### Generating yubikey private keys
+ 
+We can generate new private keys directly on the yubikey in `tsh login`. This would be ideal, since the private key will never exist outside of the yubikey and cannot be exported, and `tsh` would have oversight into the key generating process.
+
+Unfortunately, [Adding agent keys from a smartcard](https://tools.ietf.org/id/draft-miller-ssh-agent-01.html#rfc.section.4.2.5) to a user's `ssh-agent` is [not supported in x/crypto/ssh/agent](https://github.com/golang/go/issues/16304). As a result, openSSH integration and agent forwarding would not work in `tsh` with this strategy.
+
+We could potentially implement this ourselves in a fork/PR on `x/crypto/ssh/agent` or as a standalone implementation in `tsh`. It's hard to gauge how much work this would involve, but it may be worth more investigation.
```

Comment:
@Joerger I was able to work around that and looks like it's possible to implement SSH authorization without modifying Go's SSH library.